src/app/sold-products/sold-products.component.spec.ts

```diff
@@ -1,43 +1,61 @@
-import { HttpClientModule } from '@angular/common/http';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
 import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { of } from 'rxjs';
+import { SoldProducts } from '../product';
 import { ProductService } from '../product.service';
 
 import { SoldProductsComponent } from './sold-products.component';
 
 describe('SoldProductsComponent', () => {
   let component: SoldProductsComponent;
   let fixture: ComponentFixture<SoldProductsComponent>;
-  let service: ProductService;
+  let mockProductsService: ProductService;
+  let SOLD_PRODUCTS: SoldProducts; 
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      imports: [
-        HttpClientModule,
+  beforeEach( () => {
+    SOLD_PRODUCTS = {
```

Comment:
Keep mocks in separate file. It is easier to modify and read

Reply:
Updated